book/src/proposals/one-round-finality.md

```diff
@@ -0,0 +1,116 @@
+# Practical Finality in One Round of Consensus
+
+The cluster can achieve a high degree of finality though Tower
+consensus as the cluster commits to an exponentially increasing of
+safety for a block.  This safety is based on the commitment of any
+voting validator from switching votes to an alternative fork.  It
+is possible to achieve finality a different way, leaders can agree
+not to propose alternative forks.
+
+If only a part of the cluster has voted, then the set of all the
+leaders that haven't voted and the leader schedule can be used to
+simulate how lockout will grow in the future.  This calculation is
+based on the assumption that the partition is static, and that the
+same leaders will continue voting together.
+
+## Overview
+
+Leaders commit to producing a block that chains to the leaders last
+vote. Breaking this commitment is slashable.  The leader is free
+to switch forks and vote for an alternative chain.
+
+Once a block has been voted on, the subsequent leader schedule will
+reflect the likelihood of the block being finalized.
+
+For example:
+* 80% of the cluster voted for the block
+
+* The schedule for the next 5 leaders looks like:
+
+    * V,V,V, V,V,V,V, N,N,N,N, V,V,V,V, V,V,V,V
+
+Where V indicates that the leader voted on the block and N indicates
+that they haven't. By the time the `N` leader is reached, its likely
+this block would be voted on 7 times, and the lockout for this block
+would prevent the `N` leader from proposing an alternative fork
+which doesn't contain this block.
+
+## Calculating the Probability of a block succeeding
```

Comment:
I don't think section actually defines what the exact probability calcuation would be?

Reply:
@carllin this is an approximation, since the exact probability would be counting all the ways the network could reach a slot N with the block unlocked, and all the ways it can't.  So a way to approximate is do the streak calculation to see the likelihood that the V's generate enough lockout to block out the N's from creating a fork.

Reply:
In this approximation, how do you determine the percentage of the network that is in the V/N (in this example 20% was used, but how was that determined?)

Reply:
@aeyakovenko, @carllin and I were discussing this and had a few questions:

1) Can you define what a streak calculation looks like? So we have a block, then we compute the number of consecutive leaders that built on top of that block? 

2) From 1), then we can derive some measure of confidence based on this number? How does this work exactly? 

3) In your proposed computation are you factoring in the chance of a growing number of streaks? So is the probability computation you listed the chance you get a streak of 2, then a streak of some b > 2, then a streak of some c > b, until we get a root? The computation you listed only calculates the chance of at least 1 streak of length n, not a series of streaks. Is that also significant?

Reply:
1. See notes at the bottom

2, the odds of getting a longer steak get better because the number of blocks to achieve it keeps doubling.

3. the compounded percentage is the odds of all those streaks occurring consecutively, which is 93%.  This is a bit conservative since there are other sequences in which the block doesn’t fail.